jf_agent/main.py

```diff
@@ -537,22 +514,25 @@ def download_data(
     return download_data_status
 
 
-def send_data(outdir, s3_uri_prefix, aws_access_key_id, aws_secret_access_key):
-    def _s3_cmd(cmd):
-        try:
-            subprocess.check_call(
-                f'AWS_ACCESS_KEY_ID={aws_access_key_id} '
-                f'AWS_SECRET_ACCESS_KEY={aws_secret_access_key} '
-                f'{cmd}',
-                shell=True,
-                stdout=subprocess.DEVNULL,
-            )
-        except Exception:
-            print(f'ERROR: aws command failed ({cmd}) -- bad credentials?')
-            raise
+def send_data(config, creds):
+    output_basedir, timestamp = os.path.split(config.outdir) 
+
+    def get_signed_url(filename):
+        base_url = config.jellyfish_api_base
+        headers = {'Jellyfish-API-Token': creds.jellyfish_api_token}
+        r = requests.get(f'{base_url}/endpoints/agent/signed-url?filename={filename}&timestamp={timestamp}', headers=headers).json()
+        signed_url = r["signedUrl"]
+        path_to_obj = r['objectPath']
+        return signed_url, path_to_obj
+    
+    def upload_file(filename, signed_url, path_to_obj):
+        with open(f'{config.outdir}/{filename}', 'rb') as f:
+            # If successful, returns HTTP status code 204
+            upload_resp = requests.post(signed_url['url'], data=signed_url['fields'], files={'file': (path_to_obj, f)})
+            print(f'File {filename} upload HTTP status code: {upload_resp.status_code}')
 
     # Compress any not yet compressed files before sending
-    for fname in glob(f'{outdir}/*.json'):
+    for fname in glob(f'{config.outdir}/*.json'):
```

Comment:
Should we mimic `s3 sync` and do these in parallel?  The files can be large, seems like we'll be primarily gated by network, so having them each go in their own thread could be useful.